api/db-cdr/prep-tables/make-prep-survey.py

```diff
@@ -0,0 +1,455 @@
+        '../../sa-key.json')
+    bucket = storage_client.get_bucket(
+        'all-of-us-workbench-private-cloudsql')
+    return bucket.blob('cb_prep_tables/redcap/' + file)
+
+
+def query_topic_code(project, dataset, concept_name):
+    big_query_client = bigquery.Client.from_service_account_json(
+        "../../sa-key.json")
+    query = """
+    SELECT concept_code FROM `$project.dataset.concept`
+    WHERE concept_name = '$concept_name' and vocabulary_id = 'PPI' and
+    concept_class_id = 'Topic'
+    """.replace("$project.dataset", project + "." + dataset).replace(
+        "$concept_name", concept_name)
+    query_job = big_query_client.query(query)
+    return query_job.result()
+
+
+def open_writers_with_headers(name):
+    file_prefix = home_dir + "/" + surveys.get(name)
+    csv_controlled = open(file_prefix + "_controlled.csv", 'w')
+    csv_registered = open(file_prefix + "_registered.csv", 'w')
+    csv_all = open(file_prefix + "_all.csv", 'w')
+    controlled_writer = csv.DictWriter(csv_controlled, fieldnames=headers)
+    controlled_writer.writeheader()
+    registered_writer = csv.DictWriter(csv_registered, fieldnames=headers)
+    registered_writer.writeheader()
+    all_writer = csv.DictWriter(csv_all, fieldnames=headersAll)
+    all_writer.writeheader()
+    return controlled_writer, registered_writer, all_writer
+
+
+def write_row_all(writer, parent_id, code, name, item_type, min_val, max_val,
+                  flags, long_code_to_short_code):
+    global id_all
+    new_row = {'id': id_all, 'parent_id': parent_id, 'code': code,
+               'name': name, 'type': item_type, 'min': min_val, 'max': max_val,
+               'registered_topic_suppressed': flags[
+                   "REGISTERED_TOPIC_SUPPRESSED"],
+               'registered_question_suppressed': flags[
+                   "REGISTERED_QUESTION_SUPPRESSED"],
+               'registered_answer_bucketed': flags[
+                   "REGISTERED_ANSWERS_BUCKETED"],
+               'registered_answer_suppressed': flags[
+                   "REGISTERED_ANSWER_SUPPRESSED"],
+               'controlled_topic_suppressed': flags[
+                   "CONTROLLED_TOPIC_SUPPRESSED"],
+               'controlled_question_suppressed': flags[
+                   "CONTROLLED_QUESTION_SUPPRESSED"],
+               'controlled_answer_bucketed': flags[
+                   "CONTROLLED_ANSWERS_BUCKETED"],
+               'controlled_answer_suppressed': flags[
+                   "CONTROLLED_ANSWER_SUPPRESSED"]}
+    if item_type == 'QUESTION':
+        # we have to replace questionCode with short code in Field Annotation
+        new_row['code'] = get_short_code(code, long_code_to_short_code)
+    writer.writerow(new_row)
+    id_all += 1
+
+
+# builds the OutputControlled/OutputRegistered row that will get passed through the addRow function
+def write_row(writer, parent_id, code, name, item_type, min_value, max_value,
```

Comment:
think about splitting this into write_survey, write_question, write_topic. might make things cleaner